Fix #516: duplication of ids in HTML transformation

HelenaSabel opened the pull request:
This PR tackles issue #516: In the transformation to HTML, the `@xml:id` of `listBibl/bibl` elements was being added twice: first in the `<li>` element and then in its `<div>` child. 

My proposal is to just add a condition: If the child of `listBibl` being processed has already an `@xml:id`, do not call the `makeAnchor` template.

I am assigning as reviewers @ebeshero and @sydb because of your participation in the discussion of the original ticket.

HelenaSabel committed:
adds condition to avoid the creation of duplicated ids

HelenaSabel committed:
update of expected results

ebeshero reviewed (approved):
@HelenaSabel This looks good to me--glad we're producing valid HTML in the expected-results! :-)

sydb committed:
inconsequential changes:
Whitespace and other inconsequential changes that reviewers can summarily ignore.
